Model/Api/DiscountCodeValidation.php

```diff
@@ -319,7 +319,7 @@ public function validate()
             $parentQuote->getStore()->setCurrentCurrencyCode($parentQuote->getQuoteCurrencyCode());
 
             // get the coupon code
-            $discount_code = @$request->discount_code ?: @$request->cart->discount_code;
+            $discount_code = $request->discount_code ?? $request->cart->discount_code ?? null;
```

Comment:
This is a PHP7 feature, not backward compatible with PHP5.6.
Should we support merchants running PHP5.6 / Magento2.1?
Officially no, I assume, but do we have any old merchant stores running on PHP5.6?
@matt-thomason

Reply:
@danac-gs  @matt-thomason  
The reason I made this change is that Magento 2.1 is already on end-of-life and PHP 5.6 is pretty slow for Magento. So most of the merchants would likely upgrade to Magento 2.3 (or above) or even if they are on Magento 2.1 they would still use PHP7. I've done integration for a lot of merchants and I haven't seen any that used Magento 2.1

I can switch the syntax back to 
            $discount_code = isset($request->discount_code) ? $request->cart->discount_code : null;

But it is more verbose and I think it's now time to specify that our module is only working on PHP7 in composer.json